driver/ppm_tp.c

```diff
@@ -0,0 +1,111 @@
+#include "ppm_tp.h"
+
+const char *tp_names[] = {
+#define X(name, path) path,
+	TP_FIELDS
+#undef X
+};
+
+#ifndef __KERNEL__
+#include <string.h>
+#include "ppm_events_public.h"
+
+typedef struct {
+	ppm_sc_code sc_code;
+	ppm_tp_code tp_code;
+} sc_to_tp_map;
+
+static sc_to_tp_map ppm_sc_to_tp_table[] = {
```

Comment:
The engine will need to adapt this map (privately) to map each ppm_sc to its tracepoint (perhaps it will be able to do something smarter of course), and it's done.